lambdas/gro-convert-to-json/src/main/java/uk/gov/di/data/lep/GroConvertToJson.java

```diff
@@ -107,11 +113,13 @@ private String convertXmlDataToJson(String xmlData, String authorisationToken) {
 
             validateRecordCount(records, deathRegistrationGroup.recordCount());
 
-            var recordsWithAuth = records.stream()
-                .map(r -> new GroJsonRecordWithAuth(r, authorisationToken))
+            var recordsWithHeaders = records.stream()
+                .map(r -> new GroJsonRecordWithHeaders(r, authorisationToken, UUID.randomUUID().toString()))
                 .toList();
 
-            return objectMapper.writeValueAsString(recordsWithAuth);
+            recordsWithHeaders.forEach(record -> addAuditDataToQueue(generateAuditData(record.correlationId(), xmlData.hashCode())));
```

Comment:
I have a couple of thoughts on this
- I find .forEach a little confusing as a method (personal opinion :D) because it looks like it's a functional side effect-less method call, but it's actually only about side effects
- I think if we can shift the mapping from a list of events to individual audit calls down into a child method, we might be able to take advantage of the SQS Batch API which could save us a lot of overhead

Happy to talk about this though

Reply:
Oh nice! Yeah sure, happy to look at this

Reply:
Discussed with Ethan - SQS Batch API has a limit of 10 messages, and there doesn't seem to be a particularly nice way to split up the messages in Java like this without a new package. SQS doesn't have a rate limit, so going to see what this looks like, and if it doesn't look doable, I'll add a dependency to split up the list more nicely.

Reply:
I've pulled into a child method, I can't decide if it's much nicer 😆 but I don't really mind either way